filter_parser/src/lib.rs

```diff
@@ -0,0 +1,589 @@
+//! BNF grammar:
+//!
+//! ```text
+//! filter         = expression ~ EOF
+//! expression     = or
+//! or             = and (~ "OR" ~ and)
+//! and            = not (~ "AND" not)*
+//! not            = ("NOT" | "!") not | primary
```

Comment:
Could you please remove `!` from the syntax it was not in it in the previous parser.